airbyte-ci/connectors/metadata_service/lib/metadata_service/docker_hub.py

```diff
@@ -27,6 +27,75 @@ def get_docker_hub_auth_token() -> str:
     return token
 
 
+def get_docker_hub_headers() -> Dict | None:
+    if "DOCKER_HUB_USERNAME" not in os.environ or "DOCKER_HUB_PASSWORD" not in os.environ:
+        # If the Docker Hub credentials are not provided, we can only anonymously call the Docker Hub API.
+        # This will only work for public images and lead to a lower rate limit.
+        return {}
+    else:
+        token = get_docker_hub_auth_token()
+        return {"Authorization": f"JWT {token}"} if token else {}
+
+
+def get_docker_hub_tags_and_digests(
+    image_name: str,
+    retries: int = 0,
+    wait_sec: int = 30,
+    next_page_url: str | None = None,
+    tags_and_digests: Dict[str, str] | None = None,
+    paginate: bool = True,
+) -> Dict[str, str]:
+    """Find all released tags and digests for an image.
+
+    Args:
+        image_name (str): The image name to get tags and digest
+        retries (int, optional): The number of times to retry the request. Defaults to 0.
+        wait_sec (int, optional): The number of seconds to wait between retries. Defaults to 30.
+        next_page_url (str | None, optional): The next DockerHub page to consume. Defaults to None.
+        tags_and_digest (Dict[str, str] | None, optional): The accumulated tags and digests for recursion. Defaults to None.
+
+    Returns:
+        Dict[str, str]: Mapping of image tag to digest
+    """
+    headers = get_docker_hub_headers()
+    tags_and_digests = tags_and_digests or {}
+
+    if not next_page_url:
+        tags_url = f"https://registry.hub.docker.com/v2/repositories/{image_name}/tags"
```

Comment:
Can you link to the documentation for this?

I'm actually seeing a slightly different url [here](https://docs.docker.com/docker-hub/api/latest/#tag/repositories/paths/~1v2~1namespaces~1%7Bnamespace%7D~1repositories~1%7Brepository%7D~1tags/get), not sure what accounts for the difference.

Reply:
I did not use the API doc for it. As we have tag specific details in [`is_image_on_dockerhub`](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/metadata_service/lib/metadata_service/docker_hub.py#L51) with `/v2/repositories/{image_name}/tags/{tag_name}` I figured that according to rest best practices `/tags` should return a list of tags for this image and it worked.

Reply:
In general I'd prefer to use a documented API. Since we're already using the other one I guess it's fine.